### DISCUSSION TITLE
Instruction on building from scratch

### DISCUSSION THREAD

**Discussion by AliML111** [AminBhst/brisk#29] [General]
Hi Amin,

I'm excited about the potential of your project and would love to contribute my skills in GitHub Actions to enhance its development workflow.

To get started, could you please provide detailed instructions on how to build the project for various platforms? This information would be crucial for me to effectively set up and integrate GitHub Actions.

I'm confident that by leveraging GitHub Actions, we can automate tasks, streamline testing, and ensure a smoother development process across different platforms. I'm eager to collaborate and make a meaningful contribution to your project's success.

Please let me know if you have any questions or specific tasks in mind. I'm ready to roll up my sleeves and get to work!

Thanks for considering my contribution,
AliML111

**Comment by AminBhst** [comment]
@AliML111 Hey Ali. Thanks for reminding me about this. I really appreciate your enthusiasm to contribute to this project.
For starters, a Flutter SDK is required (v3.7.11). By running the "flutter build" command and providing it with the target platform as an argument, e.g. ```flutter build linux```, we get a ```build``` directory inside the project which contains the application binaries.
For Linux releases, all we need is to run a ```tar --create --file``` command on the bundle directory. The bundle directory is inside the build directory for Linux. However, I cannot remember its exact path but if you take a look inside the subdirectories of the build directory, I'm sure you can find it easily. In the future, we can also add a .rpm and a .deb build process as well.
If you were eager to find out more about how the .rpm build process will look like, I've already written a thorough article that goes through building .rpm packages for flutter applications. [Click here](https://medium.com/@amin.bhst/a-complete-guide-to-building-rpm-packages-for-flutter-applications-c8cbbbab8548).
The .deb build process, on the other hand, will definitely be simpler since there is already a [flutter package](https://github.com/jeffrey0606/flutter_to_debian) dedicated to building .deb packages.
For windows releases, though, we definitely need the .exe build process. [This youtube video](https://www.youtube.com/watch?v=XvwX-hmYv0E) can give you an idea about what needs to be done. The app that is used in this video is called "inno setup compiler" and in the end, I believe it generates a script that handles building the .exe file. There may also be other easier solutions that I'm not aware of.
That's all there is to it for now. Currently I have not written any tests but that's something that we can add to the workflow in the future. I have invited you to become a project collaborator so that you can have access to the project's GitHub Actions.
Let me know if you have any other questions.
Thank you for your contributions.

**Comment by AliML111** [comment]
@AminBhst Hi again! Sorry for my late response I had a lot going on. I managed to come up with a workflow for your project linux releases. It can be run on pushes, pull requests, and using a "Run Workflow" button in Actions GUI. There are many ways to trigger the workflow so let me know which triggers you prefer, then I will pull request it and can be used for future releases.
Also, Github supports macOS and Windows on its runners so you can continue your releases on these platforms without any problem.

**Comment by AminBhst** [comment]
Hi Ali. That's amazing thank you. I think the "Run Workflow" button would be the best approach. I cannot thank you enough. You made my life a lot easier :)

**Reply by AliML111** [reply]
Your app on linux can now be released via actions! Wait  for my pull request!

**Comment by AliML111** [comment]
What's up man? I managed to build the package for both debian based and redhat based  distros using a workflow! Now users can install Brisk without being worried about the dependencies! You can give it a try in my fork of your project:
https://github.com/AliML111/brisk
Now this leaves 2 more platforms.

**Comment by AminBhst** [comment]
Hey Ali how is it going? That's honestly SICK AF man :fire:. Amazing work. thanks a lot. I'll definitely check it out.

**Reply by AliML111** [reply]
All you need to do is to accept my invitation since the repo is private.

**Reply by AliML111** [reply]
We can still have the traditional method alongside with these packages to let users run Brisk on unsupported distros.

**Reply by AminBhst** [reply]
I just tried the RPM package on my Fedora distro and it works perfectly. What I love about it is how it says what dependencies are needed. That's perfect.
Yeah we definitely need the traditional tar file as well to support all distros.

**Reply by AminBhst** [reply]
Hey @AliML111 how are you doing?
I wanted to ask whether the .deb and .rpm build instructions are ready or not since the last time I tried them they worked really well, so I thought maybe you forgot to open a pull request for them. That's why I'm checking in. Let me know whenever you had the chance. Thanks
